client/src/navigation.js

```diff
@@ -33,36 +35,41 @@ const tabBarConfiguration = {
 };
 
 // tabs in main screen
-const MainScreenNavigator = TabNavigator({
-  Home: { screen: Home },
-  Groups: { screen: Groups },
-  Schedules: { screen: Schedules },
-  Events: { screen: Events },
-  Settings: { screen: Settings },
-}, tabBarConfiguration);
-
-const AppNavigator = StackNavigator({
-  Main: { screen: MainScreenNavigator },
-  Signin: { screen: Signin },
-  NewGroup: { screen: NewGroup },
-  SearchGroup: { screen: SearchGroup },
-  Group: { screen: Group },
-  Event: { screen: EventDetail },
-}, {
-  mode: 'modal',
-});
+const MainScreenNavigator = TabNavigator(
+  {
+    Home: { screen: Home },
+    Groups: { screen: Groups },
+    Schedules: { screen: Schedules },
+    Events: { screen: Events },
+    Settings: { screen: Settings },
+  },
+  tabBarConfiguration,
+);
+
+const AppNavigator = StackNavigator(
+  {
+    Main: { screen: MainScreenNavigator },
+    // Signin: { screen: Signin },
+    NewGroup: { screen: NewGroup },
+    SearchGroup: { screen: SearchGroup },
+    Group: { screen: Group },
+    Event: { screen: EventDetail },
+  },
+  {
+    mode: 'modal',
+  },
+);
 
 // reducer initialization code
 const firstAction = AppNavigator.router.getActionForPathAndParams('Main');
 const tempNavState = AppNavigator.router.getStateForAction(firstAction);
-const initialNavState = AppNavigator.router.getStateForAction(
-  tempNavState,
-);
+const initialNavState = AppNavigator.router.getStateForAction(tempNavState);
 
 // reducer code
 export const navigationReducer = (state = initialNavState, action) => {
   let nextState;
   switch (action.type) {
+    /*
```

Comment:
This is the way I normally load login state - separate stack for authentication rather than forcing the signin route. 

const AppWithNavigationState = (props) => {
  const { dispatch, nav } = props;

  if (props.auth.loading) {
    return <Container />;
  }

  if (!props.auth.username) {
    return <StackAuth />;
  }

  return <AppNavigator navigation={addNavigationHelpers({ dispatch, state: nav })} />;
};

Reply:
👍 
just make sure you remove all the dead code - if your solution is better lets make it hard for someone to reintroduce the older way of doing things.